src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetGenerator.cs

```diff
@@ -784,19 +784,16 @@ public static InstructionSet LookupPlatformIntrinsicInstructionSet(TargetArchite
             if (metadataType == null)
                 return InstructionSet.ILLEGAL;
 
-            string namespaceName;
+            string namespaceName = metadataType.Namespace;
             string typeName = metadataType.Name;
             string nestedTypeName = null;
-            if (metadataType.ContainingType != null)
+            while (metadataType.ContainingType != null)
             {
                 var enclosingType = (MetadataType)metadataType.ContainingType;
                 namespaceName = enclosingType.Namespace;
-                nestedTypeName = metadataType.Name;
+                nestedTypeName = nestedTypeName is null ? metadataType.Name : $""{metadataType.Name}_{nestedTypeName}"";
```

Comment:
Is `nestedTypeName` used anywhere?

Reply:
Yes, that's part of the code that is output in [CorInfoInstructionSet.cs](https://github.com/dotnet/runtime/pull/109210/files#diff-47b7c76a34a29a49750b1591fcf7ed6196296eb075ca71cc160b03ff5e717f49), where it's used to match the types nested under each top-level managed intrinsic class name.

Reply:
There seems to be a minor issue with the construction of `nestedTypeName`. The assignment should use `typeName` instead of `metadataType.Name` to correctly prepend the names of nested types. This ensures that the innermost type is properly represented first.

```suggestion
                nestedTypeName = nestedTypeName is null ? typeName : $"{typeName}_{nestedTypeName}";
```

Reply:
Those are equivalent.  I used `metadataType.Name` because the original code did, but they have the same value at this point.